drivers/spi/spi_ll_stm32.c

```diff
@@ -53,186 +53,176 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
-		blk_cfg.source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		blk_cfg->source_address = (uint32_t)&dummy_rx_tx_buffer;
+		blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
 	} else {
-		blk_cfg.source_address = (uint32_t)buf;
+		blk_cfg->source_address = (uint32_t)buf;
 		if (data->dma_tx.src_addr_increment) {
-			blk_cfg.source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+			blk_cfg->source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
 		} else {
-			blk_cfg.source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+			blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
 		}
 	}
 
-	blk_cfg.dest_address = (uint32_t)LL_SPI_DMA_GetRegAddr(cfg->spi);
+	blk_cfg->dest_address = (uint32_t)LL_SPI_DMA_GetRegAddr(cfg->spi);
 	/* fifo mode NOT USED there */
 	if (data->dma_tx.dst_addr_increment) {
-		blk_cfg.dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+		blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
 	} else {
-		blk_cfg.dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+		blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
 	}
 
 	/* give the fifo mode from the DT */
-	blk_cfg.fifo_mode_control = data->dma_tx.fifo_threshold;
+	blk_cfg->fifo_mode_control = data->dma_tx.fifo_threshold;
 
 	/* direction is given by the DT */
-	stream->dma_cfg.head_block = &blk_cfg;
-	/* give the client data as arg, as the callback comes from the dma */
+	stream->dma_cfg.head_block = blk_cfg;
+	/* give the client dev as arg, as the callback comes from the dma */
 	stream->dma_cfg.user_data = data;
 	/* pass our client origin to the dma: data->dma_tx.dma_channel */
-	ret = dma_config(data->dev_dma_tx, data->dma_tx.channel,
+	ret = dma_config(data->dma_tx.dma_dev, data->dma_tx.channel,
 			&stream->dma_cfg);
 	/* the channel is the actual stream from 0 */
 	if (ret != 0) {
 		return ret;
 	}
 
-	/* starting this dma transfer */
-	data->dma_tx.transfer_complete = false;
-
 	/* gives the request ID to the dma mux */
-	return dma_start(data->dev_dma_tx, data->dma_tx.channel);
+	return dma_start(data->dma_tx.dma_dev, data->dma_tx.channel);
 }
 
 static int spi_stm32_dma_rx_load(const struct device *dev, uint8_t *buf,
 				 size_t len)
 {
 	const struct spi_stm32_config *cfg = DEV_CFG(dev);
 	struct spi_stm32_data *data = DEV_DATA(dev);
-	struct dma_block_config blk_cfg;
+	struct dma_block_config *blk_cfg;
 	int ret;
 
 	/* retrieve active RX DMA channel (used in callback) */
 	struct stream *stream = &data->dma_rx;
 
+	blk_cfg = &stream->dma_blk_cfg;
+
 	/* prepare the block for this RX DMA channel */
-	memset(&blk_cfg, 0, sizeof(blk_cfg));
-	blk_cfg.block_size = len;
+	memset(blk_cfg, 0, sizeof(struct dma_block_config));
+	blk_cfg->block_size = len;
+
 
 	/* rx direction has periph as source and mem as dest. */
-	blk_cfg.dest_address = (buf != NULL) ? (uint32_t)buf : (uint32_t)NULL;
-	blk_cfg.source_address = (uint32_t)LL_SPI_DMA_GetRegAddr(cfg->spi);
-	if (data->dma_rx.src_addr_increment) {
-		blk_cfg.source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+	if (buf == NULL) {
+		/* if rx buff is null, then write data to dummy address. */
```

Comment:
Nice you could fix that

Reply:
not sure what you exactly mean ? The comment ? or the code ?

Reply:
the fact that rx buf can be NULL and you handle that case, afaik that was not working before with DMA enabled.

Reply:
Oh I read that wrong I thought "Nice if you could fix that" :-)